src/brevitas_examples/stable_diffusion/main.py

```diff
@@ -185,48 +283,98 @@ def bit_width(module):
+    add_bool_arg(
+        parser, 'is-sd-xl', default=False, help='Enable this flag to correctly export SDXL')
     parser.add_argument(
         '--export-target',
         type=str,
         default='',
         choices=['', 'torchscript', 'onnx'],
         help='Target export flow.')
+    add_bool_arg(
+        parser,
+        'export-weight-q-node',
+        default=False,
+        help='Enable export of floating point weights + QDQ rather than integer weights + DQ')
     parser.add_argument(
         '--conv-weight-bit-width', type=int, default=8, help='Weight bit width. Default: 8.')
     parser.add_argument(
-        '--linear-weight-bit-width', type=int, default=8, help='Weight bit width. Default: 4.')
+        '--linear-weight-bit-width', type=int, default=8, help='Weight bit width. Default: 8.')
+    parser.add_argument(
+        '--conv-input-bit-width', type=int, default=8, help='Input bit width. Default: 8.')
+    parser.add_argument(
+        '--linear-input-bit-width', type=int, default=8, help='Input bit width. Default: 8.')
     parser.add_argument(
         '--weight-param-method',
         type=str,
         default='stats',
         choices=['stats', 'mse'],
         help='How scales/zero-point are determined. Default: stats.')
+    parser.add_argument(
+        '--input-param-method',
+        type=str,
+        default='stats',
+        choices=['stats', 'mse'],
+        help='How scales/zero-point are determined. Default: stats.')
     parser.add_argument(
         '--weight-scale-precision',
         type=str,
         default='float_scale',
         choices=['float_scale', 'po2_scale'],
         help='Whether scale is a float value or a po2. Default: float_scale.')
+    parser.add_argument(
+        '--input-scale-precision',
+        type=str,
+        default='float_scale',
+        choices=['float_scale', 'po2_scale'],
+        help='Whether scale is a float value or a po2. Default: float_scale.')
     parser.add_argument(
         '--weight-quant-type',
         type=str,
         default='asym',
         choices=['sym', 'asym'],
         help='Weight quantization type. Default: asym.')
+    parser.add_argument(
+        '--input-quant-type',
+        type=str,
+        default='asym',
+        choices=['sym', 'asym'],
+        help='Input quantization type. Default: asym.')
     parser.add_argument(
         '--weight-quant-format',
         type=quant_format_validator,
         default='int',
         help=
         'Weight quantization type. Either int or eXmY, with X+Y==weight_bit_width-1. Default: int.')
+    parser.add_argument(
+        '--input-quant-format',
+        type=quant_format_validator,
+        default='int',
+        help=
+        'Weight quantization type. Either int or eXmY, with X+Y==weight_bit_width-1. Default: int.')
     parser.add_argument(
         '--weight-quant-granularity',
         type=str,
-        default='per_group',
+        default='per_channel',
```

Comment:
could you explain a bit why this changes from per group to per channel?

Reply:
We have more robust support for per_channel quantization, including export representation.

We can export per_group quantized network but we're still defining the best way to represent that quantization style.